cg/meta/upload/scout/mip_config_builder.py

```diff
@@ -0,0 +1,172 @@
+
+LOG = logging.getLogger(__name__)
+
+
+class MipConfigBuilder(ScoutConfigBuilder):
+    def __init__(
+        self,
+        hk_version_obj: hk_models.Version,
+        analysis_obj: models.Analysis,
+        mip_analysis_api: MipAnalysisAPI,
+        lims_api: LimsAPI,
+        madeline_api: MadelineAPI,
+    ):
+        super().__init__(
+            hk_version_obj=hk_version_obj, analysis_obj=analysis_obj, lims_api=lims_api
+        )
+        self.case_tags: CaseTags = CaseTags(**MIP_CASE_TAGS)
+        self.sample_tags: SampleTags = SampleTags(**MIP_SAMPLE_TAGS)
+        self.load_config: MipLoadConfig = MipLoadConfig(track="rare")
+        self.mip_analysis_api: MipAnalysisAPI = mip_analysis_api
+        self.lims_api: LimsAPI = lims_api
+        self.madeline_api: MadelineAPI = madeline_api
+
+    def build_load_config(self, rank_score_threshold: int = 5) -> None:
+        """Create a MIP specific load config for uploading analysis to Scout"""
+        LOG.info("Generate load config for mip case")
+
+        self.add_mandatory_info_to_load_config()
+        mip_analysis_data: dict = self.mip_analysis_api.get_latest_metadata(
+            self.analysis_obj.family.internal_id
+        )
+        self.load_config.human_genome_build = (
+            "38" if "38" in mip_analysis_data.get("genome_build", "") else "37"
+        )
+        self.load_config.rank_score_threshold = rank_score_threshold
+        self.load_config.rank_model_version = mip_analysis_data.get("rank_model_version")
+        self.load_config.sv_rank_model_version = mip_analysis_data.get("sv_rank_model_version")
+
+        self.load_config.gene_panels = (
+            self.mip_analysis_api.convert_panels(
+                self.analysis_obj.family.customer.internal_id, self.analysis_obj.family.panels
+            )
+            or None
+        )
+
+        self.include_case_files()
+
+        LOG.info("Building samples")
+        sample: models.FamilySample
+        for sample in self.analysis_obj.family.links:
+            self.load_config.samples.append(self.build_config_sample(sample_obj=sample))
+        self.include_pedigree_picture()
+
+    def include_pedigree_picture(self) -> None:
+        if self.is_multi_sample_case(self.load_config):
+            if self.is_family_case(self.load_config):
+                svg_path: Path = self.run_madeline(self.analysis_obj.family)
+                self.load_config.madeline = str(svg_path)
+            else:
+                LOG.info("family of unconnected samples - skip pedigree graph")
+        else:
+            LOG.info("family of 1 sample - skip pedigree graph")
+
+    def build_config_sample(self, sample_obj: models.FamilySample) -> ScoutMipIndividual:
+        """Build a sample with mip specific information"""
+
+        sample = ScoutMipIndividual()
+        self.add_mandatory_sample_info(sample=sample, sample_obj=sample_obj)
+        sample.father = sample_obj.father.internal_id if sample_obj.father else "0"
```

Comment:
can this be set automatically now? As default value "0" if not assgned

Reply:
Still need to check if father exists right?